File: client/ts/src/util/telem.ts
// Copyright 2025 Synnax Labs, Inc.
//
// Use of this software is governed by the Business Source License included in the file
// licenses/BSL.txt.
//
// As of the Change Date specified in that file, in accordance with the Business Source
// License, use of this software will be governed by the Apache License, Version 2.0,
// included in the file licenses/APL.txt.

import { type DataType, type TypedArray } from "@synnaxlabs/x";

export const randomSeries = (length: number, dataType: DataType): TypedArray => {
  // create random bytes of the correct length
  const bytes = new Uint8Array(length * dataType.density.valueOf());
  for (let i = 0; i < bytes.byteLength; i++) bytes[i] = Math.floor(Math.random() * 256);
  return new dataType.Array(bytes.buffer);
};

File: synnax/pkg/api/telem.go
// Copyright 2023 Synnax Labs, Inc.
//
// Use of this software is governed by the Business Source License included in the file
// licenses/BSL.txt.
//
// As of the Change Date specified in that file, in accordance with the Business Source
// License, use of this software will be governed by the Apache License, Version 2.0,
// included in the file licenses/APL.txt.

package api

import (
	"github.com/synnaxlabs/alamos"
	"github.com/synnaxlabs/synnax/pkg/distribution/framer"
)

type TelemService struct {
	alamos.Instrumentation
	authProvider
	Framer *framer.Service
}

func NewTelemService(p Provider) *TelemService {
	return &TelemService{
		Instrumentation: p.Instrumentation,
		Framer:          p.Config.Framer,
		authProvider:    p.auth,
	}
}
